source/develop/release-management/features/storage/discard-after-delete.html.md

```diff
@@ -0,0 +1,100 @@
+---
+authors: ishaby
+wiki_title: Features/DiscardAfterDelete
+feature_name: Discard after delete
+feature_modules: vdsm
+feature_status: Partially implemented in oVirt 4.0
+---
+
+# Discard after delete
+
+## Summary
+
+### How thinly provisioned luns were used so far
+Generally, the underlying storage can provide two types of luns - *pre allocated* and *thinly provisioned*.
+
+* When a *pre allocated* lun is created with a size of *x* Gigabytes, the storage array allocates those *x* Gigabytes for that lun, so that other luns are not able to use those blocks.<br/>
+* When a *thinly provisioned* lun is created with a size of *x* Gigabytes, the storage array may allocate less than *x* Gigabytes for that lun, and each time it reaches a certain limit of used space, it is extended by the array until it reaches it's virtual size and cannot be extended anymore.
+
+### What can be improved
+Suppose that a VM contains a disk *disk1* on a block storage domain *sd1*, and that *disk1* is removed. Also suppose that *sd1* is built on a 20GB lun, *lun1*.
+
+* If *lun1* is a *pre allocated* lun, then there's no reason to go and try to tell the storage array that some blocks were freed because *lun1* uses the whole 20GB anyway.
+* If *lun1* is a *thinly provisioned* lun, and we could tell the storage array that some blocks were freed, it could have shrinked *lun1* so that other luns in the array could make use of those unused blocks.
+
+This feature is about freeing (discarding) the whole disk space right before removing it.
+
+### For whom is this feature useful
+This feature is useful for users that have thinly provisioned luns and wish to reclaim space when removing a disk from a block storage domain that is built from them.
+
+
+## Owner
+* Idan Shaby
+* Email: <ishaby@redhat.com>
+
+
+## Current status
+* [Phase one](#usage) (implemented in vdsm config file) - released in oVirt 4.0.
+* [Phase two](#future-plans) (configurable on engine side) - planned for oVirt 4.1.
+
+
+## General functionality and restrictions
+
+### Under the hood
+A new parameter was added to the *vdsm* configuration file, ***discard_enable***.<br/>
```

Comment:
I'll explain it in the usage part.
Technically it was added to the config.py.in file but the user should use the /etc/vdsm/vdsm.conf file.